src/control/fold.lean

```diff
@@ -352,13 +352,16 @@ end
 
 variables {m : Type u → Type u} [monad m] [is_lawful_monad m]
 
+section
+local attribute [semireducible] opposite
```

Comment:
This is not a pretty fix, but this issue is too mplicated for me to understand.  git tells me you wrote this, @cipher1024.